lizmap/modules/lizmap/classes/lizmapWMSRequest.class.php

```diff
@@ -185,7 +185,7 @@ protected function getcontext()
             array('repository' => $this->repository->getKey(), 'project' => $this->project->getKey())
         );
         $sUrl = str_replace('&', '&amp;', $sUrl).'&amp;';
-        $data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $data);
+        $response->data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $response->data);
```

Comment:
```suggestion
        $data = $response->data;
        $data = preg_replace('/xlink\:href=".*"/', 'xlink:href="'.$sUrl.'&amp;"', $data);
```
I prefer to work as immutable even if it is not.
And change the line 93